### PR TITLE
Fix 2 x panic: runtime error: index out of range

### PR DESCRIPTION
**Pull request by QuentinPerez**
```
/Users/quentinperez/go/src/github.com/derekparker/delve/proc/proc_darwin.go:39 +0x8ad
/Users/quentinperez/go/src/github.com/derekparker/delve/service/debugger/debugger.go:57 +0x37d
```

```
/Users/quentinperez/go/src/github.com/derekparker/delve/cmd/dlv/main.go:243 +0x264
```

### CODE REVIEW

File: cmd/dlv/main.go [hunk 1]
```diff
@@ -106,6 +107,12 @@ starts and attaches to it, and enables you to immediately begin debugging your p
 	execCommand := &cobra.Command{
 		Use:   "exec [./path/to/binary]",
 		Short: "Runs precompiled binary, attaches and begins debug session.",
+		PersistentPreRunE: func(cmd *cobra.Command, args []string) error {
+			if len(args) == 0 {
+				return errors.New("You must specified a ./path/to/binary")
```
Comment (derekparker) [inline cmd/dlv/main.go]: Reword error as: "you must provide a path to a binary".

File: cmd/dlv/main.go [hunk 2]
```diff
@@ -239,6 +246,12 @@ starts and attaches to it, and enable you to immediately begin debugging your pr
 		Use:   "attach [pid]",
 		Short: "Attach to running process and begin debugging.",
 		Long:  "Attach to running process and begin debugging.",
+		PersistentPreRunE: func(cmd *cobra.Command, args []string) error {
+			if len(args) == 0 {
+				return errors.New("You must specified a PID")
```
Comment (derekparker) [inline cmd/dlv/main.go]: Reword as: "you must provide a PID".

**Comment by derekparker** [comment]
Apparently there is a bug in the vendored version of cobra, can you please update it and verify this patch works with the vendored version. You can compile normally once the new cobra is vendored, just make sure you have set the environment variable for the Go vendor experiment: `$ export GO15VENDOREXPERIMENT=1`.

**Comment by QuentinPerez** [comment]
This patch works with the vendored version of cobra 😉

**Comment by derekparker** [comment]
Are you sure? Here's my output:

```
derek at delve(QuentinPerez-fix_panic) make install                                                                        {2}
go install  github.com/derekparker/delve/cmd/dlv
derek at delve(QuentinPerez-fix_panic) dlv attach
panic: runtime error: invalid memory address or nil pointer dereference
[signal 0xb code=0x1 addr=0x0 pc=0x513930]

goroutine 1 [running]:
github.com/derekparker/delve/vendor/github.com/spf13/cobra.(*Command).execute(0xc8200a0e00, 0xd05a20, 0x0, 0x0, 0x0, 0x0)
	/home/derek/code/go/src/github.com/derekparker/delve/vendor/github.com/spf13/cobra/command.go:512 +0x530
github.com/derekparker/delve/vendor/github.com/spf13/cobra.(*Command).Execute(0xc8200a01c0, 0x0, 0x0)
	/home/derek/code/go/src/github.com/derekparker/delve/vendor/github.com/spf13/cobra/command.go:617 +0x46a
main.main()
	/home/derek/code/go/src/github.com/derekparker/delve/cmd/dlv/main.go:286 +0xc8d

goroutine 17 [syscall, locked to thread]:
runtime.goexit()
	/usr/local/go/src/runtime/asm_amd64.s:1696 +0x1

goroutine 5 [syscall]:
os/signal.loop()
	/usr/local/go/src/os/signal/signal_unix.go:22 +0x18
created by os/signal.init.1
	/usr/local/go/src/os/signal/signal_unix.go:28 +0x37
derek at delve(QuentinPerez-fix_panic) dlv exec                                                                            {2}
panic: runtime error: invalid memory address or nil pointer dereference
[signal 0xb code=0x1 addr=0x0 pc=0x513930]

goroutine 1 [running]:
github.com/derekparker/delve/vendor/github.com/spf13/cobra.(*Command).execute(0xc82013c700, 0xd05a20, 0x0, 0x0, 0x0, 0x0)
	/home/derek/code/go/src/github.com/derekparker/delve/vendor/github.com/spf13/cobra/command.go:512 +0x530
github.com/derekparker/delve/vendor/github.com/spf13/cobra.(*Command).Execute(0xc82013c000, 0x0, 0x0)
	/home/derek/code/go/src/github.com/derekparker/delve/vendor/github.com/spf13/cobra/command.go:617 +0x46a
main.main()
	/home/derek/code/go/src/github.com/derekparker/delve/cmd/dlv/main.go:286 +0xc8d

goroutine 17 [syscall, locked to thread]:
runtime.goexit()
	/usr/local/go/src/runtime/asm_amd64.s:1696 +0x1

goroutine 5 [syscall]:
os/signal.loop()
	/usr/local/go/src/os/signal/signal_unix.go:22 +0x18
created by os/signal.init.1
	/usr/local/go/src/os/signal/signal_unix.go:28 +0x37
derek at delve(QuentinPerez-fix_panic)
```

I think this commit https://github.com/spf13/cobra/commit/72ee37f33edf9acbb30f15510eec8958fd84da19 fixes it, which was pulled in 9 days ago.

Are you compiling with the Go 1.5 vendor experiment enabled?

**Comment by QuentinPerez** [comment]
```console
~/go/src/github.com/derekparker/delve -> fix_panic $ rm -rf Godeps
~/go/src/github.com/derekparker/delve -> fix_panic $ export GO15VENDOREXPERIMENT=1
~/go/src/github.com/derekparker/delve -> fix_panic $ godep save ./...
~/go/src/github.com/derekparker/delve -> fix_panic $ CERT=dlv-cert make install
go install -ldflags="-s" github.com/derekparker/delve/cmd/dlv
codesign -s dlv-cert /Users/quentinperez/go/bin/dlv
~/go/src/github.com/derekparker/delve -> fix_panic $ /Users/quentinperez/go/bin/dlv attach
Usage:
  dlv attach [pid] [flags]

Flags:
  -h, --help[=false]: help for attach

Global Flags:
      --headless[=false]: Run debug server only, in headless mode.
      --init="": Init file, executed by the terminal client
  -l, --listen="localhost:0": Debugging server listen address.
      --log[=false]: Enable debugging server logging.

Error: you must provide a PID

~/go/src/github.com/derekparker/delve -> fix_panic $ git diff Godeps/Godeps.json
                {
                        "ImportPath": "github.com/peterh/liner",
-                       "Rev": "941195f85dd7a23971a3d55142972d396278ddcf"
+                       "Rev": "b850cf8c6d0ee52309aad09ac610508c6c75e819"
                },
                {
                        "ImportPath": "github.com/russross/blackfriday",
@@ -33,7 +33,7 @@
                },
                {
                        "ImportPath": "github.com/spf13/cobra",
-                       "Rev": "53d96508f5c9311f5dade714e1de7bd6a9d2af4a"
+                       "Rev": "d732ab3a34e6e9e6b5bdac80707c2b6bad852936"
                },
                {
```

I do a mistake ?

**Comment by derekparker** [comment]
> I do a mistake ?

No, but you did update the vendored copy before you compiled :). In any event, I'm going to put a commit on top of this that updates the vendored cobra version, and also fixes a few other things with the vendor dir (apparently godeps was vendoring test code from other packages).

Thanks for the patch, I'll merge it in shortly.

**Comment by QuentinPerez** [comment]
Thanks you 😊

**Pull request merged**